include/seqan3/alignment/pairwise/align_result_selector.hpp

```diff
@@ -52,7 +52,7 @@ struct align_result_selector
 {
 private:
     //!\brief The user configured score type.
-    using score_type = typename alignment_configuration_traits<configuration_t>::original_score_type;
+    using score_type = typename alignment_configuration_traits<remove_cvref_t<configuration_t>>::original_score_type;
```

Comment:
Why `cvref`? `std::remove_reference_t` should be enough, if you just want to allow `alignment_configuration_traits` to access `typename configuration_t::my_own_type`.

Reply:
I run into problems when someone removed the `const` from the type and I could access a `non-const member` in a concept, but got a hard-compiler error when I actual couldn't access the member.

I would prefer to ALWAYS only remove the reference when you try to access the type.

See
https://godbolt.org/z/CpFqpw

Reply:
Yes, that is true. I saw it like enable_view were I am only allowed to overload it for cv-unqualified types. But in case of a traits class I might want to preserve constness to define the correct types. That is of course not the case with the alignment traits right now but who knows.